File: PROGRAMMERS/힙_이중우선순위큐.js
function solution(operations) {
  let problem = operations
  let answer = [];

  let result = [];

  for (let i = 0; i < problem.length; i++) {
    if (result && problem[i] === 'D 1') {
      result.pop()
    } else if (result && problem[i] === 'D -1') {
      result.shift()
    } else {
      let a = problem[i].split(' ');
      let num = parseInt(a[1])
      result.push(num)
      console.log(result)
      result.sort(function(a, b) {
        return a - b
      })
    }
  }
  if (result.length > 1) {
    answer = [result[result.length-1], result[0]]
  } else {
    answer = [0,0]
  }

  return answer;
}


console.log(solution(	["I -45", "I 653", "D 1", "I -642", "I 45", "I 97", "D 1", "D -1", "I 333"]))

File: PROGRAMMERS/힙_이중우선순위큐.py

def solution(operations):
    answer = []

    result = []

    for i in range(len(operations)):
        if result and operations[i] == 'D 1':
            result.pop()
        elif result and operations[i] == 'D -1':
            result.pop(0)
        else:
            a, b = operations[i].split()
            alpha = a
            num = int(b)
            if alpha == 'I':
                result.append(num)
            result.sort()

    if result:
        answer = [result[-1], result[0]]
    else:
        answer = [0,0]

    return answer

print(solution(["I 7","I 5","I -5","D -1"]))